doc/specs/#597 - Tab Sizing.md

```diff
@@ -0,0 +1,95 @@
+---
+author: Kayla Cinnamon @cinnamon-msft
+created on: 2020-01-03
+last updated: 2020-01-03
+issue id: 597
+---
+
+# Tab Sizing
+
+## Abstract
+
+This spec outlines the tab sizing feature. This is an application-level feature that is not profile-specific (at least for now).
+
+Global properties that encompass tab sizing:
+
+* `tabWidthMode` (accepts pre-defined values for tab sizing behavior)
+* `tabWidthMin` (can never be smaller than the icon width)
+* `tabWidthMax` (can never be wider than the tab bar)
+
+Acceptable values for `tabWidthMode`:
+
+* [default] `equal` (all tabs are sized the same, regardless of tab title length)
+* `titleLength` (width of tab contains entire tab title)
+
+## Inspiration
+
+Other browsers and terminals have varying tab width behavior, so we should give people options.
+
+## Solution Design
+
+`tabWidthMode` will be a global setting that will accept the following strings:
+
+* `equal`
+    * All tabs are equal in width
+    * If the tab bar has filled, tabs will shrink as additional tabs are added
+    * Utilizes the `equal` setting from WinUI's TabView
+
+* `titleLength`
+    * Tab width varies depending on title length
+    * Width of tab will fit the whole tab title
+    * Utilizes the `sizeToContent` setting from WinUI's TabView
+
+In addition to `tabWidthMode`, the following global properties will also be available:
+
+* `tabWidthMin`
+    * Accepts an integer
+    * Value correlates to the minimum amount of pixels the tab width can be
+    * If value is less than the width of the icon, the minimum width will be the width of the icon
+    * If value is greater than the width of the tab bar, the maximum width will be the width of the tab bar
+    * If not set, the tab will have the system-defined minimum width
+
+* `tabWidthMax`
+    * Accepts an integer
+    * Value correlates to the maximum amount of pixels the tab width can be
+    * If value is less than the width of the icon, the minimum width will be the width of the icon
+    * If value is greater than the width of the tab bar, the maximum width will be the width of the tab bar
+    * If not set, the tab will have the system-defined maximum width

```

Comment:
which is?

Reply:
After a bit of testing, looks like XAML doesn't set a max width for us:
![really long tab title](https://user-images.githubusercontent.com/11050425/83201460-5f569780-a0fa-11ea-922b-859afba1ab43.png)

We should keep this behavior as default (aka 'infinite').

EDIT: ok, _theoretically_ I could keep going but I'm just assuming there's no max width...